### PR TITLE
refactor: refact metadata store

### PR DESCRIPTION
**Pull request by rudyfly**
### Ⅰ. Describe what this PR did
Refact metadata store, merge volume's store and container's store, add
boltdb into meta package, and optimizate the store's bucket.

### Ⅴ. Special notes for reviews

Signed-off-by: Rudy Zhang <rudyflyzhang@gmail.com>

### CODE REVIEW

File: pkg/meta/boltdb.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	b.db = db
+
+	return nil
+}
+
+func (b *bolt) prepare(db *boltdb.DB, bucket []byte) error {
+	return db.Update(func(tx *boltdb.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(bucket)
+		if err != nil {
+			return errors.Wrap(err, "boltdb: create bucket")
```
Comment (allencloud) [inline pkg/meta/boltdb.go]: If that, the error string would be `boltdb: create bucket: err.Error()`. I am wondering if we could make it like `failed to create bucket in boltdb: err.Error()`, so the format should be `return errors.Wrap(err, "failed to create bucket in boltdb")` @rudyfly

File: pkg/meta/boltdb.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+// Put is used to put metadate into boltdb.
+func (b *bolt) Put(bucket, key string, value []byte) error {
+	return b.db.Update(func(tx *boltdb.Tx) error {
+		bkt := tx.Bucket([]byte(bucket))
+		if bkt == nil {
+			return ErrBucketNotFound
+		}
+		if err := bkt.Put([]byte(key), value); err != nil {
+			return errors.Wrap(err, "boltdb: put key: "+key)
```
Comment (allencloud) [inline pkg/meta/boltdb.go]: Two much colon, the result error string would be `boltdb: put key: sdfgh: err.Error()`, how about `failed to put key asdfgh in boltdb: err.Error()` using `errors.Wrap(err, fmt.Sprintf("failed to put key %s in boltdb", key))`

File: pkg/meta/store.go [hunk 3]
```diff
@@ -47,9 +48,21 @@ type Store struct {
 
 // NewStore creates a backend storage.
 func NewStore(cfg Config) (*Store, error) {
+	if len(cfg.Buckets) < 1 {
+		return nil, fmt.Errorf("config bucket can not be nil")
```
Comment (allencloud) [inline pkg/meta/store.go]: s/nil/empty

`empty` is a user's perspective and nil is only a coder's perspective.

File: pkg/meta/store.go [hunk 4]
```diff
@@ -47,9 +48,21 @@ type Store struct {
...
+	}
+
+	if cfg.Driver == "" {
+		cfg.Driver = DefaultStore
+	}
+
+	if _, ok := backend[cfg.Driver]; !ok {
+		return nil, fmt.Errorf("can not found store driver: %s", cfg.Driver)
```
Comment (allencloud) [inline pkg/meta/store.go]: return nil, fmt.Errorf("store driver %s not found", cfg.Driver)

File: pkg/meta/store_test.go [hunk 5]
```diff
@@ -26,6 +26,7 @@ func (d *Demo2) Key() string {
 var s *Store
 
 var cfg = Config{
+	Driver:  "local",
```
Comment (allencloud) [inline pkg/meta/store_test.go]: Could you add unit test for store driver "boltdb"?

File: pkg/meta/boltdb.go (hunk 1 above)
Comment (rudyfly) [reply pkg/meta/boltdb.go]: ok

File: pkg/meta/store_test.go (hunk 5 above)
Comment (rudyfly) [reply pkg/meta/store_test.go]: ok

**Comment by rudyfly** [comment]
@allencloud PTAL

**Comment by allencloud** [comment]
LGTM, thanks a lot for your work. This is fantastic pull request help us remove the redundant code and test cases cover what we add quite well. 🥂 🌞

**Pull request merged**